### ISSUE TITLE
Long running query during import process(more than 10.hrs)

### ISSUE THREAD

**Issue by MALKARAJ** [osm-search/Nominatim#3315]
**Software Environment (please complete the following information):**

Nominatim version: v4.2.3
Postgresql version: 15
Postgis version: 3.3.3

**Hardware Configuration (please complete the following information):**

nominatim import machine

RAM: 64GB
number of CPUs: 32CORE
type and size of disks: nvme and 1.5TB

`UPDATE word SET info = info __ jsonb_build_object__ count) FROM word_frequencies WHERE word_id = id` this query ran for more than 10.5 hour in postgres and cause memory out of bound issue

**Comment by mtmail** [comment]
With a dozen questions you're overusing the free community support. If your server runs out of memory you can add more memory (hardware). The installation instructions are clear that 64GB isn't enough.

**Comment by MALKARAJ** [comment]
> With a dozen questions you're overusing the free community support. If your server runs out of memory you can add more memory (hardware). The installation instructions are clear that 64GB isn't enough.

you are doing great work by answering the questions really fast and accurate , appreciating your time and efforts, sorry for overasking the question


running postgres and import process as a separate machine

**Here is the postgres configuration**
```

RAM: 128GB
number of CPUs: 64CORE
type and size of disks: nvme and 1.5TB

```

**Comment by lonvia** [comment]
I've seen this query run for a very long time on our production machines as well but can't reproduce it on my development server. This indicates that the queries themselves are not really the issue. My guess is that we are running into a forced vacuum again. The query updates a lot of lines in the word table. Lets see if it helps to create a new word table instead.

I don't think the memory out of bound issues are directly Nominatim's fault. Sounds more like a misconfiguration of Postgres and I tend to agree with @mtmail here that you have used up your quota for these kind of generic support questions.

**Comment by lonvia** [comment]
Closed by #3328
